src/main/java/com/google/javascript/clutz/PlatformSymbols.java

```diff
@@ -0,0 +1,1716 @@
+package com.google.javascript.clutz;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+
+/**
+ * There are many "platform" symbols, where Closure defines them in an externs file and TypeScript
+ * defines them in a lib.d.ts file. When a Closure file references a platform type like HTMLElement
+ * or platform global like encodeURIComponent, rather than referring to a type that Clutz derives
+ * from the externs we'd rather it refer directly to the TypeScript type.
+ *
+ * <p>Note that doing this is not strictly correct (the two may have slightly different definitions)
+ * but this is likely what users expect and it reduces the output size.
+ *
+ * <p>This class maintains all the metadata about these types; for example, Closure and TypeScript
+ * have different names for the interface that is the 'style' property on an element, so this class
+ * has a map from Closure name to TypeScript name.
+ */
+public class PlatformSymbols {
+  /**
+   * Symbols that are mapped to a different specific name. E.g. on the JS side when it says
+   * "CSSProperties", it's referring to the TypeScript type "CSSStyleDeclaration".
+   */
+  public static final ImmutableMap<String, String> CLOSURE_TO_TYPESCRIPT =
+      new ImmutableMap.Builder<String, String>()
+          .put("CSSProperties", "CSSStyleDeclaration")
+          .build();
+
+  /**
+   * List of global platform symbols that are redirected through an alias in closure.lib.d.ts This
+   * allows the following pattern to work:
+   *
+   * <pre>
+   * namespace foo {
+   *   class Error extends Error {}
+   * }
+   * </pre>
+   *
+   * by replacing the second Error with GlobalError.
+   *
+   * <p>Closure has internal "externs" for some global symbols:
+   * https://github.com/google/closure-compiler/blob/master/src/com/google/javascript/rhino/jstype/JSTypeRegistry.java
+   * Which will be emitted in each file that references them in partial mode. For example, using
+   * Date in a file results in:
+   *
+   * <pre>
+   * declare namespace ಠ_ಠ.clutz {
+   *   class Date extends Date_Instance {
+   *   }
+   *   class Date_Instance {
+   *     private noStructuralTyping_: any;
+   *     constructor (a ? : any , b ? : any , c ? : any , d ? : any , e ? : any , f ? : any , g ? : any ) ;
+   *   }
+   * }
+   * </pre>
+   *
+   * Adding a symbol to this list also suppresses the duplicate externs.
+   */
+  public static final ImmutableSet<String> GLOBAL_SYMBOL_ALIASES =
+      ImmutableSet.of("Error", "Event", "EventTarget", "Object", "Date");
+
+  /**
+   * Set of all symbols that are defined in a platform externs.js that we don't have matching
+   * TypeScript type for. We could in theory just let Clutz generate types for these, but in
+   * practice no users have needed them yet and it saves file space. It should be safe to remove
```

Comment:
Sentence repeated in the top of the class. I think it is not needed here.

Reply:
Whcih sentence?

Reply:
nevermind, I thought that 'not strictly correct' and 'in theory' referred to the same exception. Now I see I was wrong.